File: FTAND03U2C1/app/src/main/java/com/example/ft_and_03u_2c_1/MainActivity.java
package com.example.ft_and_03u_2c_1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button etSignIn, etSignUp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        etSignIn=findViewById(R.id.BtnSignInPage);
        etSignUp=findViewById(R.id.BtnSignUpPage);

        etSignIn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, SignInActivity.class);
                startActivity(intent);
            }
        });
        etSignUp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this,SignUpActivity.class);
                startActivity(intent);
            }
        });
    }


}

File: AndroidshorttakeawayassignmentI/app/src/main/java/com/example/androidshorttake_awayassignment_i/MainActivity.kt
package com.example.androidshorttake_awayassignment_i

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.widget.doAfterTextChanged
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import com.application.searchapi.data.Status
import com.example.androidshorttake_awayassignment_i.data.model.Address
import com.example.androidshorttake_awayassignment_i.databinding.ActivityMainBinding
import com.example.androidshorttake_awayassignment_i.repository.Repo
import com.example.androidshorttake_awayassignment_i.ui.Adapter.SearchAdapter
import com.example.androidshorttake_awayassignment_i.viewModel.SearchViewModel
import com.example.androidshorttake_awayassignment_i.viewModel.ViewModelFactory
import dagger.hilt.android.AndroidEntryPoint
import kotlin.properties.Delegates

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    private lateinit var adapter: SearchAdapter
    private var list = emptyList<Address>()
   lateinit var  viewModel:SearchViewModel

//    val viewModel:SearchViewModel by viewModel
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val repo = Repo()

        val viewModelFactory = ViewModelFactory(repo)
        viewModel = ViewModelProviders.of(this, viewModelFactory).get(SearchViewModel::class.java)

        binding.etSearch.doAfterTextChanged {
            if(binding.etSearch.text.toString().length > 0){
                binding.tvWelcome.visibility = View.GONE
                binding.rcRecyclerView.visibility = View.VISIBLE
                loadData(binding.etSearch.text.toString())
            }else{
                binding.rcRecyclerView.visibility = View.GONE
                binding.tvWelcome.visibility = View.VISIBLE
            }

        }



    }

    fun loadData(string: String) {
        viewModel.getListAddress(string).observe(this, {
            when (it.status) {
                Status.ERROR -> {

                }
                Status.SUCCESS -> {

                    list = it.data?.data?.addressList!!
                    adapter = SearchAdapter(list)
                    val linearLayoutManager = LinearLayoutManager(this)
                    binding.apply {
                        rcRecyclerView.adapter = adapter
                        rcRecyclerView.layoutManager = linearLayoutManager
                    }
                    adapter.notifyDataSetChanged()
                }
            }
        })
    }

}